cocos/audio/apple/AudioPlayer.mm

```diff
@@ -122,6 +122,23 @@ of this software and associated documentation files (the "Software"), to deal
                 delete _rotateBufferThread;
                 _rotateBufferThread = nullptr;
                 ALOGVV("rotateBufferThread exited!");
+
+#if CC_TARGET_PLATFORM == CC_PLATFORM_IOS
+                // some specific OpenAL implement defects existed on iOS platform
+                // refer to: https://github.com/cocos2d/cocos2d-x/issues/18597
+                ALint sourceState;
+                ALint bufferProcessed = 0;
+                alGetSourcei(_alSource, AL_SOURCE_STATE, &sourceState);
+                if (sourceState == AL_PLAYING) {
+                    alGetSourcei(_alSource, AL_BUFFERS_PROCESSED, &bufferProcessed);
+                    while (bufferProcessed < QUEUEBUFFER_NUM) {
+                        std::this_thread::sleep_for(std::chrono::milliseconds(2));
```

Comment:
I have counted time cost of this code block, it's about `2 * QUEUEBUFFER_TIME_STEP`, about cost 160-240 ms when the QUEUEBUFFER_TIME_STEP is `0.1`, about 80-120 ms when the QUEUEBUFFER_TIME_STEP is `0.05`. 

not matter when changing the sleep time to 1 ms or 2 ms. change the QUEUEBUFFER_TIME_STEP from 0.1 to 0.05 will half the time cost. It's meaningful. 

after changing `QUEUEBUFFER_TIME_STEP` to `0.05`, I have tested play bigger audio file on iPhone 5s, 6, 8, all play smoothly.